pycbc/results/pygrb_postprocessing_utils.py

```diff
@@ -395,134 +427,145 @@ def get_bestnrs(trigs, q=4.0, n=3.0, null_thresh=(4.25, 6), snr_threshold=6.,
+        ifo_snr[ifo] = trigs['snr_%s' % ifo]
     # Apply this cut only if there is more than 1 IFO
     if len(ifos) > 1:
-        for i_trig, _ in enumerate(trigs):
+        for i_trig in range(len(snr)):
             # Apply only to triggers that were not already cut previously
             if bestnr[i_trig] != 0:
                 ifos.sort(key=lambda ifo, j=i_trig: sens[ifo][j], reverse=True)
                 if (ifo_snr[ifos[0]][i_trig] < sngl_snr_threshold or
                         ifo_snr[ifos[1]][i_trig] < sngl_snr_threshold):
                     bestnr[i_trig] = 0
-    for i_trig, trig in enumerate(trigs):
+
+    for i_trig in range(len(bestnr)):
         # Get chisq reduced (new) SNR for triggers that were not cut so far
-        # NOTE: .get_bestnr is in glue.ligolw.lsctables.MultiInspiralTable
+        # Ideally we should not have any of them right now
+        # NOTE: it may be redundant without the implementation of bank and
+        # cont chisq
         if bestnr[i_trig] != 0:
-            bestnr[i_trig] = trig.get_bestnr(index=q, nhigh=n,
-                                             null_snr_threshold=null_thresh[0],
-                                             null_grad_thresh=null_grad_thresh,
-                                             null_grad_val=null_grad_val)
+            # Define the input trig as a dictionary
+            trig = {
+                'snr': trigs['coherent_snr'][i_trig],
+                'null_snr': trigs['null_snr'][i_trig],
+                'chisq': trigs['chisq'][i_trig],
+                'nifos': trigs['nifos'][i_trig],
+                    }
             # If we got this far and the bestNR is non-zero, verify that chisq
             # was actually calculated for the trigger, otherwise raise an
             # error with info useful to figure out why this happened.
-            if bestnr[i_trig] != 0 and trig.chisq == 0:
+            if bestnr[i_trig] != 0 and trig['chisq'] == 0:
                 err_msg = "Chisq not calculated for trigger with end time "
-                err_msg += f"{trig.get_end()} and SNR {trig.snr}."
+                err_msg += f"{time[i_trig]} and SNR {trig['snr']}."
                 raise RuntimeError(err_msg)
 
     return bestnr
 
 
+# =============================================================================
+# Veto trigger
+# =============================================================================
+def veto_trig(trigs, trial_dict):
+    """Remove trigs that are not in trial dict"""
+    mask = []
+    for time in trigs['end_time']:
+        mask.append(any([time in trial for trial in trial_dict]))
+    return {key: trigs[key][mask] for key in trigs.keys()}
+
+
 # =============================================================================
 # Construct sorted triggers from trials
 # =============================================================================
 def sort_trigs(trial_dict, trigs, slide_dict, seg_dict):
```

Comment:
I've already modified this function to work directly using the HDF5 files with the efficiency code.